trailblazer/store/base.py

```diff
@@ -61,6 +61,10 @@ def get_filtered_analyses(self, analyses: Query, query: AnalysesRequest) -> Quer
             filters.append(AnalysisFilter.FILTER_BY_STATUSES)
         if query.type:
             filters.append(AnalysisFilter.FILTER_BY_TYPES)
+        if query.case_id:
+            filters.append(AnalysisFilter.FILTER_BY_CASE_ID)
+        if query.most_recent:
+            filters.append(AnalysisFilter.FILTER_BY_MOST_RECENT)
```

Comment:
can we break this function up instead of adding to this? It seems like this can end up with an amount of if statements proportionate to the number of attributes an analyses can be filtered on. Or at least, let's not keep adding to this.

How do we now at any given point in a function call what is actually being filtered on when we apply this function without needing to check what the `query` actually has stored.

Also, `analyeses: Query` and `query: AnalysesRequest` is extremely confusing